### PR TITLE
Adapt default+schedule for YaST online cases for x86_64 part 2

### PR DESCRIPTION
**Pull request by hjluo**
Adapt default+schedule for YaST online cases for x86_64 part 2.

- Related ticket:  https://progress.opensuse.org/issues/119887
- Related MR:     https://gitlab.suse.de/qsf-y/qa-sle-functional-y/-/merge_requests/455
- Needles:           N/A
- Verification run:  [YaST_Online_Part2](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP5&build=hjluo%2Fos-autoinst-distri-opensuse%23x86_def_online_p2)

### CODE REVIEW

**Comment by jknphy** [comment]
Let's remove `+flow` from title/pr as we don't use flows.
We should not convert scenarios that are not installations or do not use the atomicity achieved with libyui-rest-api, so also discarded Multi-machine ones or offline. Please take a look to this.

@hjluo Consider in case the reviewers will find more than a few differences between original schedules and new ones, to perform a diff of both before continue applying feedback.

**Comment by hjluo** [comment]
> Let's remove `+flow` from title/pr as we don't use flows. We should not convert scenarios that are not installations or do not use the atomicity achieved with libyui-rest-api, so also discarded Multi-machine ones or offline. Please take a look to this.

OK. changed.
 
> @hjluo Consider in case the reviewers will find more than a few differences between original schedules and new ones, to perform a diff of both before continue applying feedback.
@jknphy  Do you mean to put the diff in the code review? for all these files are mostly newly 
created so we can't list the diff on the code review page.

File: schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  except for "base" and "minimal". Not to be confused with the new system role "minimal" introduced with SLE15.
+vars:
+  DEPENDENCY_RESOLVER_FLAG: 1
+  DESKTOP: textmode
+  PATTERNS: base,enhanced_base
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
```
Comment (jknphy) [inline schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml:14]: this is repeated, it is already in default.

**Comment by jknphy** [comment]
> > Let's remove `+flow` from title/pr as we don't use flows. We should not convert scenarios that are not installations or do not use the atomicity achieved with libyui-rest-api, so also discarded Multi-machine ones or offline. Please take a look to this.
> 
> OK. changed.
> 
> > @hjluo Consider in case the reviewers will find more than a few differences between original schedules and new ones, to perform a diff of both before continue applying feedback.
> 
> @jknphy Do you mean to put the diff in the code review? for all these files are mostly newly created so we can't list the diff on the code review page.

I was thinking about to make a diff and show here the diff between original schedule and new one with a link, but I'm not sure because I review a bit your PR and as you already have experience you are not making mistakes that could be detected by a diff, there are subtle things case by case. Therefore let's wait for the other reviewers and hope they can check line by line and find something. The problem is that one line here changed creates a different test coverage, so we need to be careful.

**Comment by jknphy** [comment]
@JRivrain @tinawang123

File: schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+description:    >
+  Select a minimal textmode installation by starting with the default and unselecting all patterns
+  except for "base" and "minimal". Not to be confused with the new system role "minimal" introduced with SLE15.
+vars:
+  DEPENDENCY_RESOLVER_FLAG: 1
+  DESKTOP: textmode
+  PATTERNS: base,enhanced_base
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
```
[the remark this answers is not in the document]
Comment (hjluo) [reply schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml:12]: changed.

File: schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml (hunk 1 above)
Comment (hjluo) [reply schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml:14]: removed

File: schedule/yast/textmode/textmode_x86_64.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  YUI_REST_API: 1
+schedule:
+  system_preparation:
+    - console/consoletest_setup
+  system_validation:
+    - console/validate_product_installed_SLES
+    - console/verify_network
+    - locale/keymap_or_locale
+    - console/validate_installed_patterns
+    - console/force_scheduled_tasks
```
Comment (jknphy) [inline schedule/yast/textmode/textmode_x86_64.yaml:15]: looks like the test are not in order in the original job, could you order them, please? this one looks preparation not validation.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  EXPECTED_INSTALL_HOSTNAME: dhcphostname
+  EXTRABOOTPARAMS: ifcfg=*=dhcp
+  NICTYPE_USER_OPTIONS: hostname=dhcphostname
+  YUI_REST_API: 1
+schedule:
+  local_user: []
+  booting: []
+  default_systemd_target: []
+  installation_settings:
+    - installation/hostname_inst
```
Comment (jknphy) [inline schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:18]: let's hook this with timezone and put it the first, so the below we will have only empty hooks for which we can create some feature later to avoid writing them. Please apply the same in the other *hostname* test suites.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+name: yast_hostname+dhcp_hostname
+description: >
+  Test suite uses default option to set dhcp (using ifgf=*=dhcp), hostname.
+  Test validates if installation can successfully start in case of usage of
+  these parameters.
+vars:
+  EXIT_AFTER_START_INSTALL: '1'
```
Comment (jknphy) [inline schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:8]: this is not needed with yaml schedule. Please correct in the other *hostname* test suites.

**Review by jknphy — COMMENTED** [review]
LGTM other than those small changes to address.

File: schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
+name:           minimal+base@yast
+description:    >
+  Select a minimal textmode installation by starting with the default and unselecting all patterns
+  except for "base" and "minimal". Not to be confused with the new system role "minimal" introduced with SLE15.
+vars:
+  DEPENDENCY_RESOLVER_FLAG: 1
```
[the remark this answers is not in the document]
Comment (hjluo) [reply schedule/yast/minimal+base/minimal+base@yast_x86_64.yaml:7]: dropped.

File: schedule/yast/minimal+role_minimal/minimal+role_minimal_x86_64.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+---
+name: minimal+role_minimal
+description: >
+  SLE 15 specific registered installation with minimal role selected.
+  Installation is validated by successful boot and that YaST does not report
+  any issue.
+vars:
+  BSC1167736: '1'
+  SYSTEM_ROLE: minimal
```
[the remark this answers is not in the document]
Comment (hjluo) [reply schedule/yast/minimal+role_minimal/minimal+role_minimal_x86_64.yaml:9]: removed.

File: schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  Installation where we modify some pre-existing partitions. Must depend on some
+  create_hdd test suite.
+vars:
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/skip_module_registration
+  add_on_product:
+    - installation/add_on_product/skip_install_addons
+  suggested_partitioning:
```
[the remark this answers is not in the document]
Comment (hjluo) [reply schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml:13]: changed and removed the upper 2 keys.

File: schedule/yast/msdos/msdos_x86_64.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+---
+name:           msdos
+description:    >
+  Test for installation on msdos partition table.
+vars:
+  YUI_REST_API: 1
+schedule:
+  suggested_partitioning:
```
[the remark this answers is not in the document]
Comment (hjluo) [reply schedule/yast/msdos/msdos_x86_64.yaml:8]: changed.

File: schedule/yast/textmode/textmode_x86_64.yaml (hunk 3 above)
Comment (hjluo) [reply schedule/yast/textmode/textmode_x86_64.yaml:15]: changed and ordered.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml (hunk 5 above)
Comment (hjluo) [reply schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:8]: removed.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml (hunk 4 above)
Comment (hjluo) [reply schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:18]: changed.

File: schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+---
+name: modify_existing_partition
+description: >
+  Installation where we modify some pre-existing partitions. Must depend on some
+  create_hdd test suite.
+vars:
+  YUI_REST_API: 1
+schedule:
+  expert_partitioning:
```
Comment (JRivrain) [inline schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml:9]: I don't see "expert_partitioning" in the default, is it normal?

File: schedule/yast/msdos/msdos_x86_64.yaml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+---
+name:           msdos
+description:    >
+  Test for installation on msdos partition table.
+vars:
+  YUI_REST_API: 1
+schedule:
+  expert_partitioning:
```
Comment (JRivrain) [inline schedule/yast/msdos/msdos_x86_64.yaml:8]: same as above

File: schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  Guided Partitioning installation with ext4 filesystem.
+vars:
+  FILESYSTEM: ext4
+  YUI_REST_API: 1
+schedule:
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  guided_partitioning:
```
Comment (JRivrain) [inline schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml:13]: I guess there is a reason, but why the change ? In guided_btrfs and guided_ext4 we don't have these extra steps. I mean, it works so it's ok, but I would just have copied the same.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+  EXPECTED_INSTALL_HOSTNAME: dhcphostname
+  EXTRABOOTPARAMS: ifcfg=*=dhcp
+  NICTYPE_USER_OPTIONS: hostname=dhcphostname
+  YUI_REST_API: 1
+schedule:
+  local_user: []
+  booting: []
+  clock_and_timezone:
+    - installation/clock_and_timezone/accept_timezone_configuration
+    - installation/hostname_inst
```
Comment (JRivrain) [inline schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:17]: I would put it in local_user, but ok.

File: schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml (hunk 10 above)
Comment (hjluo) [reply schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml:9]: expert_partitioning is in the part1 PR, but I added it here

File: schedule/yast/msdos/msdos_x86_64.yaml (hunk 11 above)
Comment (hjluo) [reply schedule/yast/msdos/msdos_x86_64.yaml:8]: added.

File: schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml (hunk 13 above)
Comment (hjluo) [reply schedule/yast/yast_hostname/yast_hostname+dhcp_hostname_x86_64.yaml:16]: changed to local_user.

File: schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml (hunk 12 above)
Comment (hjluo) [reply schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml:13]: you mean: select_guided_setup,accept_default_part_scheme,accept_default_fs_options in [guided_btrfs](https://openqa.nue.suse.com/tests/10436392) , [guided_xfs](https://openqa.nue.suse.com/tests/10436394) and [guided_ext4](https://openqa.nue.suse.com/tests/10436393) they all have these 3 modules.

File: schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml (hunk 10 above)
Comment (JRivrain) [reply schedule/yast/modify_existing_partition/modify_existing_partition_x86_64.yaml:9]: Ah, ok, I did not know about the other PR.

File: schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml (hunk 12 above)
Comment (JRivrain) [reply schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml:13]: Now I'm really confused: where are those modules called? for example in btrfs, those modules are not in the [schedule file](https://openqa.nue.suse.com/tests/10436392/settings/schedule/yast/sle/guided_btrfs/guided_btrfs.yaml) but also not in the [default](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/schedule/yast/sle/flows/default.yaml). Do you know hat is going on ?

Comment (JRivrain) [reply schedule/yast/sle/guided_ext4/guided_ext4_x86_64.yaml:13]: So correct me if I'm wrong, but you ned to remove YAML_SCHEDULE_FLOWS from the VRs in order to use the new schedule file. I re-ran the VRs for btrfs, ext4 and xfs with YAML_SCHEDULE_FLOWS=''

File: schedule/yast/sle/guided_xfs/guided_xfs_x86_64.yaml [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+schedule:
+  extension_module_selection:
+    - installation/module_registration/register_module_desktop
+  system_role:
+    - installation/system_role/accept_selected_role_SLES_with_GNOME
+  guided_partitioning:
+    - installation/partitioning/select_guided_setup
+    - installation/partitioning/guided_setup/accept_default_part_scheme
+    - installation/partitioning/guided_setup/select_filesystem_option_xfs
+    - installation/partitioning/accept_proposed_layout
```
Comment (JRivrain) [inline schedule/yast/sle/guided_xfs/guided_xfs_x86_64.yaml:16]: @jknphy Is [this](https://openqa.suse.de/tests/10458327#step/accept_proposed_layout#1/2) expected? I thought a section in schedule file was overriding the default, but here it seems like what is in default is appended at the end.

Comment (JRivrain) [inline schedule/yast/sle/guided_xfs/guided_xfs_x86_64.yaml:16]: This module is already scheduled in [suggested_partitioning](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/60b576bbb7a8e930486b9754bfabf66dce07c0ed/schedule/yast/sle/flows/default.yaml#L27). Also, there is no guided_partitioning in the default file, it's a bug that it works in my opinion.

Comment (hjluo) [reply schedule/yast/sle/guided_xfs/guided_xfs_x86_64.yaml:16]: changed.
